src/chart/funnel/funnelLayout.ts

```diff
@@ -277,9 +322,14 @@ export default function funnelLayout(ecModel: GlobalModel, api: ExtensionAPI) {
         }
 
         const funnelAlign = seriesModel.get('funnelAlign');
-        let gap = seriesModel.get('gap');
-        const viewSize = orient === 'horizontal' ? viewWidth : viewHeight;
-        let itemSize = (viewSize - gap * (data.count() - 1)) / data.count();
+        let viewSize: number;
+        if (dynamicHeight) {
+            viewSize = orient === 'vertical' ? viewWidth : viewHeight;
```

Comment:
It's better to do the same check 'horizontal' so the behavior will be more consistent when the value is neither `vertical` nor `horizontal`